controllers/reconcile.go

```diff
@@ -853,6 +853,97 @@ func setupFedModules(feEnv *crd.FrontendEnvironment, frontendList *crd.FrontendL
 	return nil
 }
 
+func adjustSearchEntry(searchEntry *crd.SearchEntry, frontend crd.Frontend) crd.SearchEntry {
+	newSearchEntry := crd.SearchEntry{
+		// make the id environment and frontend specific to reduce duplicate ids across Frontend resources
+		ID:          fmt.Sprintf("%s-%s-%s", frontend.Name, frontend.Spec.EnvName, searchEntry.ID),
+		Title:       searchEntry.Title,
+		Description: searchEntry.Description,
+		Href:        searchEntry.Href,
+		AltTitle:    searchEntry.AltTitle,
+		IsExternal:  searchEntry.IsExternal,
+	}
+	return newSearchEntry
+}
+
+func setupSearchIndex(feList *crd.FrontendList) []crd.SearchEntry {
+	searchIndex := []crd.SearchEntry{}
+	for _, frontend := range feList.Items {
+		if frontend.Spec.SearchEntries != nil {
+			for _, searchEntry := range frontend.Spec.SearchEntries {
+				searchIndex = append(searchIndex, adjustSearchEntry(searchEntry, frontend))
+			}
+		}
+	}
+	return searchIndex
+}
+
+func setupWidgetRegistry(felist *crd.FrontendList) []crd.WidgetEntry {
+	widgetRegistry := []crd.WidgetEntry{}
+	for _, frontend := range felist.Items {
+		for _, widget := range frontend.Spec.WidgetRegistry {
+			widgetRegistry = append(widgetRegistry, *widget)
+		}
+	}
+
+	return widgetRegistry
+}
+
+func getServiceTilePath(section string, group string) string {
+	return fmt.Sprintf("%s-%s", section, group)
+}
+
+func setupServiceTilesData(felist *crd.FrontendList, feEnvironment crd.FrontendEnvironment) ([]crd.FrontendServiceCategoryGenerated, []string) {
+	categories := []crd.FrontendServiceCategoryGenerated{}
+	if feEnvironment.Spec.ServiceCategories == nil {
+		// skip if we do not have service categories
+		return categories, []string{}
+	}
+
+	// just a quick cache to make it easier and faster to assign tiles to their destination
+	tileGroupAccessMap := make(map[string]*[]crd.ServiceTile)
+
+	for _, category := range *feEnvironment.Spec.ServiceCategories {
+		groups := []crd.FrontendServiceCategoryGroupGenerated{}
+		for _, gr := range category.Groups {
+			tiles := []crd.ServiceTile{}
+			group := crd.FrontendServiceCategoryGroupGenerated{
+				ID:    gr.ID,
+				Title: gr.Title,
+				Tiles: &tiles,
```

Comment:
I *think* you can remove the pointer here because slices are already reference types. You might be able to do just `Titles: titles` but I'm not 100% sure. Might be fun to find out!

Reply:
I think I already figured it out. And I can't I need the reference later and without the pointer the slice was not updated and the output was always an empty array.